inc/io.php

```diff
@@ -271,41 +305,45 @@ function io_deleteFromFile($file,$badline,$regex=false){
     // load into array
     if(substr($file,-3) == '.gz'){
         $lines = gzfile($file);
+    }else if(substr($file,-4) == '.bz2'){
+        $lines = bzfile($file, true);
     }else{
         $lines = file($file);
     }
 
     // remove all matching lines
     if ($regex) {
-        $lines = preg_grep($badline,$lines,PREG_GREP_INVERT);
+        if($maxlines > 0) {
+            $matches = preg_grep($oldline, $lines);
+            $count = 0;
+            foreach($matches as $ix=>$m) {
+                $lines[$ix] = preg_replace($oldline, $newline, $m);
+                if(++$count >= $maxlines) break;
+            }
+        } else {
+            $lines = ($maxlines == 0) ? preg_grep($oldline, $lines, PREG_GREP_INVERT)
+                                      : preg_replace($oldline, $newline, $lines, $maxlines);
+        }
     } else {
-        $pos = array_search($badline,$lines); //return null or false if not found
```

Comment:
couldn't this block be merged into the above code by appropriate quoting/escaping of $oldline & $newline to make them into regexes.  e.g. ```
$pattern = $regex ? $oldline : '/'.preg_quote($oldline, '/').'/';
$replacement = $regex ? $newline : addcslashes($newline,'$\');
``` 
[ not tested, but you get the idea ]